sd-dev/bin/begin.py

```diff
@@ -1,41 +1,36 @@
-import argparse
+#!/usr/bin/env python3
+
+import json
 import os
-import requests
 import subprocess
 import shutil
 
-def parse_args():
-    """
-    Handle CLI args.
-    """
-    parser = argparse.ArgumentParser()
-    parser.add_argument(
-        "--commit",
-        default=False,
-        required=True,
-        action="store",
-        help="Git commit to build from",
-    )
-    args = parser.parse_args()
-    return args
-
-def run(commit):
+def run():
     owner = "freedomofpress"
     repo = "securedrop-workstation"
 
     working_dir = "/var/lib/sdci-ci-runner"
     subprocess.run(["sudo", "mkdir", "-p", working_dir])
     subprocess.run(["sudo", "chown", "user:user", working_dir])
 
-    workspace = f"{repo}_{commit}"
+    # Read context in from file
+    context_filepath = "/home/user/QubesIncoming/dom0/context.json"
+    if os.path.exists(context_filepath):
+        with open(context_filepath, "r") as context_file:
+            context = json.load(context_file)
+    else:
+        raise SystemError("context.json file not found!")
+    commit_sha = context.get("commit", False)
```

Comment:
Previously `--commit` was required, did you mean to make it optional here? Then we end up with the possibility of `workspace = "securedrop-workstation_False"`

Reply:
I'm not following you, using `.get()` is what makes them optional?

```
>>> print({}.get('foo'))
None
>>> {}['foo']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'foo'
```